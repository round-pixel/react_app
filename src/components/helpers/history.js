import { browserHistory as history } from 'react-router-dom';

export default history;
