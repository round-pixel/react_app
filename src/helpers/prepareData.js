import { compact } from 'lodash/array';
import { map } from 'lodash/collection';

export default function(store, state) {
  const { location, query, params, routes } = state;
  const prepareDataFns = compact(map(routes, route => route.prepareData));
  
  return map(
    prepareDataFns,
    prepareData => prepareData(store, query, params, location)
  );
}
