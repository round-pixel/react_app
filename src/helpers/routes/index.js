export const newPostPath = () => '/new';
export const editPostPath = (id = ':id') => `/posts/${id}/edit`;
export const postsPath = (id = ':id') => `/posts/${id}`;
export const rootPath = () => '/';
export const aboutPath = () => '/about';
export const contactPath = () => '/contact';
