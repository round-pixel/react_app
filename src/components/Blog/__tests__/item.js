import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { fakeStore } from 'helpers/fakeStore';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter } from 'react-router-dom';

configure({ adapter: new Adapter() });

import Item from '../Card';

describe('Item', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store  = fakeStore({});
    ReactDOM.render(
      <Provider store={store}>
        <MemoryRouter>
          <Item />
        </MemoryRouter>
      </Provider>,
      div
    );
  });
  describe('render', () => {
    it('should render usual item', () => {
      const itemProps = { message: 'Hello World', image: {}, metaInfo: { createdBy: 'Bob'}, id: 5 };
      const item = shallow(<Item props={itemProps}/>);

      expect(item).toMatchSnapshot();
    });
  });
});
