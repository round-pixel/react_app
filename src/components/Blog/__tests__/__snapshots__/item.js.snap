// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render should render usual item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(MyCard)
    props={
      Object {
        "id": 5,
        "image": Object {},
        "message": "Hello World",
        "metaInfo": Object {
          "createdBy": "Bob",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "avatar": "MyCard-avatar-5",
        "card": "MyCard-card-1",
        "expand": "MyCard-expand-3",
        "expandOpen": "MyCard-expandOpen-4",
        "flexGrow": "MyCard-flexGrow-6",
        "likeCount": "MyCard-likeCount-7",
        "media": "MyCard-media-2",
      },
      "id": 0,
      "image": Object {
        "src": "/images/river.jpg",
      },
      "metaInfo": Object {},
      "props": Object {
        "id": 5,
        "image": Object {},
        "message": "Hello World",
        "metaInfo": Object {
          "createdBy": "Bob",
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "avatar": "MyCard-avatar-5",
          "card": "MyCard-card-1",
          "expand": "MyCard-expand-3",
          "expandOpen": "MyCard-expandOpen-4",
          "flexGrow": "MyCard-flexGrow-6",
          "likeCount": "MyCard-likeCount-7",
          "media": "MyCard-media-2",
        },
        "id": 0,
        "image": Object {
          "src": "/images/river.jpg",
        },
        "metaInfo": Object {},
        "props": Object {
          "id": 5,
          "image": Object {},
          "message": "Hello World",
          "metaInfo": Object {
            "createdBy": "Bob",
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
